gazebosim/gz-transport
# Oneway service request from the command line

# 🦟 Bug fix

Partially fixes #97

## Summary


This patch makes it possible to request a one-way service request from the command line. Before, it was possible passing `--reptype gz.msgs.Empty` but always getting the message `Service call timed out` as an output (because the service callback wasn't answering anything, which was expected).

## How to test it?

You can compile the examples, and then:

1. Launch the one-way responser:
```
./responser_oneway
```

2. Request the service using the two-way request and verify that you don't get the timeout:
```
gz service -s /oneway --reqtype gz.msgs.StringMsg --reptype gz.msgs.Empty --req 'data: "Hello"' --timeout 1000
```

3. Use the new one-way request and verify that you don't get the timeout message:
```
gz service -s /oneway --reqtype gz.msgs.StringMsg --req 'data: "Hello"'
``` 

Note that both requests should reach the responser:
```
Request received: [Hello]
Request received: [Hello]
```

## Checklist

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.
